kernel/kernel-api/src/main/java/io/delta/kernel/types/CollationIdentifier.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.types;
+
+import static io.delta.kernel.internal.util.Preconditions.checkArgument;
+
+import io.delta.kernel.annotation.Evolving;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Identifies collation for string type. <a
+ * href="https://github.com/delta-io/delta/blob/master/protocol_rfcs/collated-string-type.md#collation-identifiers">
+ * Collation identifiers</a>
+ *
+ * @since 3.3.0
+ */
+@Evolving
+public class CollationIdentifier {
+
+  private final String provider;
+  private final String name;
+  private final Optional<String> version;
+
+  private CollationIdentifier(String provider, String collationName) {
+    this(provider, collationName, Optional.empty());
+  }
+
+  private CollationIdentifier(String provider, String collationName, Optional<String> version) {
+    Objects.requireNonNull(provider, "Collation provider cannot be null.");
+    Objects.requireNonNull(collationName, "Collation name cannot be null.");
+    Objects.requireNonNull(version, "Provider version cannot be null.");
```

Comment:
I don't think we ever specified that this is a provider version, it is just the version of the collation. Can you remove the provider reference here and in other places?